mod-source-record-manager-server/src/main/java/org/folio/verticle/JobMonitoringWatchdogVerticle.java

```diff
@@ -0,0 +1,109 @@
+import org.folio.services.JobMonitoringService;
+import org.folio.spring.SpringContextUtil;
+
+@Component
+@PropertySource("classpath:application.properties")
+public class JobMonitoringWatchdogVerticle extends AbstractVerticle {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+  private static AbstractApplicationContext springGlobalContext;
+
+  @Autowired
+  private JobMonitoringService jobMonitoringService;
+
+  @Autowired
+  private JobExecutionService jobExecutionService;
+
+  @Value("${job.monitoring.watchdog.timestamp}")
+  private long watchdogTimestamp;
+  private long timerId;
+
+  public static void setSpringContext(AbstractApplicationContext springContext) {
+    JobMonitoringWatchdogVerticle.springGlobalContext = springContext;
+  }
+
+  @Override
+  public void start() {
+    declareSpringContext();
+
+    timerId = vertx.setPeriodic(watchdogTimestamp, handler -> monitorJobExecutionProgress());
+  }
+
+  @Override
+  public void stop() throws Exception {
+    vertx.cancelTimer(timerId);
+    super.stop();
+  }
+
+  protected void declareSpringContext() {
+    context.put("springContext", springGlobalContext);
+    SpringContextUtil.autowireDependencies(this, context);
+  }
+
+  private void monitorJobExecutionProgress() {
+    getTenants().forEach(tenantId -> {
+      LOGGER.info("Check tenant [{}] for stacked jobs", tenantId);
+      jobMonitoringService.getAll(tenantId).onSuccess(jobMonitors ->
+        jobMonitors.forEach(jobMonitoring -> {
+          long lastEventTimestamp = getDifferenceTimestamp(jobMonitoring.getLastEventTimestamp());
+          if (lastEventTimestamp >= watchdogTimestamp) {
+            findJobExecutionById(tenantId, jobMonitoring);
+          }
+        })
+      );
+    });
+  }
+
+  private Set<String> getTenants() {
+    LocalMap<String, Integer> tenants = vertx.sharedData().getLocalMap("tenants");
+    return tenants.keySet();
+  }
+
+  private long getDifferenceTimestamp(Date lastEventTimestamp) {
+    return new Date().toInstant().toEpochMilli() - lastEventTimestamp.toInstant().toEpochMilli();
+  }
+
+  private void findJobExecutionById(String tenantId, JobMonitoring jobMonitoring) {
+    jobExecutionService.getJobExecutionById(jobMonitoring.getJobExecutionId(), tenantId)
+      .onSuccess(optionalJobExec -> optionalJobExec.ifPresent(jobExecution -> printWarnLog(jobExecution, tenantId)));
+  }
+
+  private void printWarnLog(JobExecution jobExecution, String tenantId) {
+    LOGGER.warn("Data Import Job with jobExecutionId = {} not progressing for tenant = {}, "
+        + "current time = {}, run by = {}, file name = {}, job profile info = {}, start date = {}, stop date = {}",
+      jobExecution.getId(), tenantId, new Date(), printRunBy(jobExecution), jobExecution.getFileName(),
+      printJobProfile(jobExecution), jobExecution.getStartedDate(), jobExecution.getCompletedDate()
+    );
+  }
+
+  private String printRunBy(JobExecution jobExecution) {
```

Comment:
Maybe better to rename to 'getRunBy' and 'getJobProfileInfo' ?

Reply:
Sure, it will be better